### PR TITLE
docs(pattern): Added Cloud PAL guidance to Common actions pattern

### PR DESCRIPTION
**Pull request by connor-leech**
Through a design jam, Cloud PAL appended Carbon's common action/CRUD patterns with additional guidance. Much of that guidance has been added with this PR.

### CODE REVIEW

File: src/pages/patterns/common-actions/index.mdx [hunk 1]
```diff
@@ -2,14 +2,11 @@
 title: Common actions
 ---
 
-### Common actions are frequently used actions that appear multiple times across different components and workflows. For platform consistency, these actions should only be applied in the ways described below.
+<PageDescription>
 
-<AnchorLinks>
+Common actions are frequently used actions that appear multiple times across different components and workflows. For platform consistency, these actions should only be applied in the ways described below.
```
Comment (janchild) [inline src/pages/patterns/common-actions/index.mdx]:
```suggestion
Common actions are frequently used actions that appear across different components and workflows. For platform consistency, these actions should only be applied in the ways described below.
```

File: src/pages/patterns/common-actions/index.mdx [hunk 2]
```diff
@@ -19,56 +16,111 @@ title: Common actions
...
 
+#### Non-dangerous deletion (~5% of cases)
+
+Use when it's trivial to undo deletion or re-create the item. Delete the item upon click or tap without further warning.
+
+#### Moderate danger deletion (~80% of cases)
+
+Use when an action cannot be undone or the item can't be easily recreated. This pattern is also useful if you're deleting more than one thing.
+
+Ask for confirmation of the delete, with guidance on what will occur if they delete.
```
Comment (janchild) [inline src/pages/patterns/common-actions/index.mdx]:
```suggestion
Ask for confirmation of the delete, with guidance about what will occur if they delete.
```

File: src/pages/patterns/common-actions/index.mdx [hunk 3]
```diff
@@ -19,56 +16,111 @@ title: Common actions
...
 
 ![Example of delete in a Modal and Overflow Menu](images/common-action-4.png)
 
+#### Non-dangerous deletion (~5% of cases)
+
+Use when it's trivial to undo deletion or re-create the item. Delete the item upon click or tap without further warning.
+
+#### Moderate danger deletion (~80% of cases)
+
+Use when an action cannot be undone or the item can't be easily recreated. This pattern is also useful if you're deleting more than one thing.
```
Comment (janchild) [inline src/pages/patterns/common-actions/index.mdx]:
```suggestion
Use when an action cannot be undone or the item cannot be recreated easily. This pattern is also useful if you're deleting more than one thing.
```

File: src/pages/patterns/common-actions/index.mdx [hunk 4]
```diff
@@ -19,56 +16,111 @@ title: Common actions
...
+
+#### Post-deletion flow
+
+After the user deletes an item, return to the page that lists the item deleted. Animate the removal of the item from the list or page and present a success notification.
+
+If the deletion fails, raise a notification to tell the user that deletion failed. Send a second notification on another communication channel, like email, if possible. Animate the item back onto the page if possible.
+
+### Errors
+
+Errors occur when an action or process does not succeed. Error notifications can occupy full pages, form fields, notifications (toast, inline, etc.), and modals. Ultimately, all should provide context of 'what happened' and a clear path to continue.
```
Comment (janchild) [inline src/pages/patterns/common-actions/index.mdx]:
```suggestion
Errors occur when an action or process does not succeed. Error notifications can occupy full pages, form fields, notifications (toast, inline, etc.), and modals. Ultimately, all should provide context of what happened and a clear path to continue.
```

File: src/pages/patterns/common-actions/index.mdx [hunk 5]
```diff
@@ -19,56 +16,111 @@ title: Common actions
...
+</Column>
+</Row>
+
+Consider redirecting the user to a previous state, a support page, or by offering helpful recommendations. Be honest and helpful.
+
+Some components, like text input and form field errors, are quite small and require more thoughtful approaches to the space and placement of error handling. Consider inline error notifications for these instances.
+
+#### Copy guidelines
+
+Be brief, honest, supportive, and human. Explain what happened and what can the user do to resolve the error?
```
Comment (janchild) [inline src/pages/patterns/common-actions/index.mdx]:
```suggestion
Be brief, honest, supportive, and human. Explain what happened and what can the user do to resolve the error.
```

File: src/pages/patterns/common-actions/index.mdx [hunk 6]
```diff
@@ -19,56 +16,111 @@ title: Common actions
...
+
+For full-page and large modals, keep error messages no longer than 3 paragraph lines. For form errors, keep error messages no longer than two lines.
```
Comment (janchild) [inline src/pages/patterns/common-actions/index.mdx]:
```suggestion
For full-page and large modals, keep error messages no longer than three paragraph lines. For form errors, keep error messages no longer than two lines.
```

File: src/pages/patterns/common-actions/index.mdx [hunk 7]
```diff
@@ -19,56 +16,111 @@ title: Common actions
...
+#### Hierarchy and placement
+
+Represent remove as a button or `subtract` icon or glyph. The remove action is rarely the primary action on the page and should not be overly emphasized.
+
+#### Considerations
+
+- What are the implications of the add action for the user? Are there financial, access, or legal considerations?
+- This action can be confused with deleting.
+- A user may not have the correct permissions for this action.
+- Inform the user if the action is permanent
```
Comment (janchild) [inline src/pages/patterns/common-actions/index.mdx]:
```suggestion
- Inform the user if the result is permanent.
```

File: src/pages/patterns/common-actions/index.mdx [hunk 8]
```diff
@@ -19,56 +16,111 @@ title: Common actions
...
+- How long will the action take? Seconds, minutes, hours, or days?
+- What should the user do if the removal fails?
+- Is this a single or bulk action?
+
 ### Reset
 
-This action reverts values back to their last saved state. The last saved state includes the values stored the last time the user clicked or triggered "Apply."
+Reset reverts values back to their last saved state. The last saved state includes the values stored the last time the user clicked or triggered "Apply."
```
Comment (janchild) [inline src/pages/patterns/common-actions/index.mdx]:
```suggestion
Reset reverts values back to their last saved state. The last saved state includes the values stored the last time the user clicked or triggered **Apply**.
```

**Comment by connor-leech** [comment]
> Are we moving common actions to experimental? The nav link doesn't work?

Good catch. We were going to bring this guidance into experimental as distinct patterns but elected to roll the relevant guidance into our production common action patterns.

**Pull request merged**